classes/Relations/Reference/Builder.php

```diff
@@ -99,6 +122,125 @@ public function setClass(string $class): self
         return $this;
     }
 
+    /**
+     * @note Does not check whether column or property exists
+     * @param string $localKeyColumn
+     * @return $this
+     */
+    private function initLocalKeyColumn(string $localKeyColumn): self
+    {
+        $this->localKeyProperty = $this->policy->properties($this->localClass)[$localKeyColumn] ?? null;
+
+        return $this;
+    }
+
+    /**
+     * @param string $localKeyColumn
+     * @return $this
+     */
+    public function setLocalKeyColumn(string $localKeyColumn): self
+    {
+        $this->localKeyProperty = $this->propertyByColumn($this->localClass, $localKeyColumn);
+
+        return $this;
+    }
+
+    /**
+     * @param Property|string $localKeyProperty
+     * @return $this
+     */
+    public function setLocalKey($localKeyProperty): self
+    {
+        if (!$localKeyProperty instanceof Property) {
```

Comment:
We might want to deprecate the ability to pass a property object.

Reply:
Deprecation added to changelog.md and implemented by removal of Property from the @param tag of both setLocalKey and setRemoteKey. A proper IDE will start warning you about passing a Property instance.

trigger_error('...', E_USER_DEPRECATED) is a little risky since I'd like to release this within the 0.11 minor version line.